### PR TITLE
*: add fcos distribution

### PR DESCRIPTION
**Pull request by arithx**
Note: The test related changes are only for tests that run on the `qemu` platform.

### CODE REVIEW

File: platform/platform.go [hunk 1]
```diff
@@ -299,14 +299,35 @@ func CheckMachine(ctx context.Context, m Machine) error {
...
 
-	if !bytes.Equal(out, []byte("ID=coreos")) && !bytes.Equal(out, []byte(`ID="rhcos"`)) {
-		return fmt.Errorf("not a supported instance")
+	supportedSystemCheck := func() (bool, error) {
+		// ensure we're talking to a supported system
+		switch string(out) {
+		case `ID=coreos`, `ID="rhcos"`:
+			return true, nil
+		case `ID=fedora`:
+			variantid, _, err := m.SSH("grep ^VARIANT_ID= /etc/os-release")
```
Comment (cgwalters) [inline platform/platform.go]: I'd do `. /etc/os-release && echo "$VARIANT_ID"` - the file is shell syntax, and your "parser" here doesn't handle quotes.

Comment (cgwalters) [reply platform/platform.go]: Also worth noting this obsoletes https://github.com/coreos/mantle/pull/920

File: kola/harness.go [hunk 2]
```diff
@@ -493,6 +493,14 @@ func scpKolet(c cluster.TestCluster, mArch string) {
 			if err := c.DropFile(kolet); err != nil {
 				c.Fatalf("dropping kolet binary: %v", err)
 			}
+			if Options.Distribution == "rhcos" || Options.Distribution == "fcos" {
+				for _, machine := range c.Machines() {
+					out, stderr, err := machine.SSH("sudo chcon -t bin_t kolet")
```
Comment (jlebon) [inline kola/harness.go]: Would be good to add some context comment here on why this is needed.

**Comment by arithx** [comment]
Updates pushed addressing comments :arrow_up:

**Comment by arithx** [comment]
Pushed addressing comments :arrow_up:

File: platform/platform.go [hunk 3]
```diff
@@ -299,14 +299,24 @@ func CheckMachine(ctx context.Context, m Machine) error {
...
-	// ensure we're talking to a Container Linux system
-	out, stderr, err := m.SSH("grep ^ID= /etc/os-release")
+	out, stderr, err := m.SSH(`. /etc/os-release && echo "$ID-$VARIANT_ID"`)
 	if err != nil {
 		return fmt.Errorf("no /etc/os-release file: %v: %s", err, stderr)
 	}
 
-	if !bytes.Equal(out, []byte("ID=coreos")) && !bytes.Equal(out, []byte(`ID="rhcos"`)) {
-		return fmt.Errorf("not a supported instance")
+	supportedSystemCheck := func() bool {
```
Comment (bgilbert) [inline platform/platform.go]: The inline function isn't needed any longer.

**Pull request merged**